src/components/geometry.js

```diff
@@ -24,6 +29,7 @@ module.exports.Component = registerComponent('geometry', {
    */
   update: function (previousData) {
     var data = this.data;
+    var diffData = diff(previousData || {}, data);
```

Comment:
Try not to use `diff` when possible, it's more expensive than just doing an equality check